components/compass_theme_provider/compass_theme_provider.tsx

```diff
@@ -1,11 +1,78 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
+import {blendColors, convertToRgb, rgbToHex, setAlpha} from '@mattermost/compass-components/shared/color-utils';
 import React, {useState, useEffect} from 'react';
+import {createGlobalStyle} from 'styled-components';
 
 import ThemeProvider, {lightTheme} from '@mattermost/compass-components/utilities/theme';
 
 import {Theme} from 'mattermost-redux/types/themes';
+import Constants from '../../utils/constants';
+import * as UserAgent from '../../utils/user_agent';
+
+const updateCodeTheme = (userTheme: string): void => {
+    let cssPath = '';
+    Constants.THEME_ELEMENTS.forEach((element) => {
+        if (element.id === 'codeTheme') {
+            element.themes?.forEach((theme) => {
+                if (userTheme === theme.id) {
+                    cssPath = theme.cssURL;
+                }
+            });
+        }
+    });
+
+    const link = document.querySelector<HTMLLinkElement>('link.code_theme');
+    if (link && cssPath !== link.href) {
+        // changeCss('code.hljs', 'visibility: hidden');
+
+        const xmlHTTP = new XMLHttpRequest();
+
+        xmlHTTP.open('GET', cssPath, true);
+        xmlHTTP.onload = function onLoad() {
+            link.href = cssPath;
+
+            if (UserAgent.isFirefox()) {
+                link.addEventListener('load', () => {
+                    // changeCss('code.hljs', 'visibility: visible');
```

Comment:
Are these intentionally commented out? I don't remember the reason for them to begin with

Reply:
I suppose it was to prevent flickering of the code blocks, when the theme is in the process of changing. I left it commented out on purpose so I can re-apply it when it proves to be wrong or ugly! :D

Reply:
Just be aware of if changing these styles causes elements to shift on the page since that would cause scrolling issues. I don't think visibility changes the position, but that's just something I'd like to mention.

Reply:
I just tried it and it seems like the change is quite quick ... I removed them and if there are any regressions we can re-enable that